Update documenation : install plugin from URL (#4799)
Napari/napari: docs/plugins/find_and_install_plugin.md

   ![napari viewer's Plugin dialog. At the bottom of the dialog, there is a place to install by name, URL, or dropping in a file.](/images/plugin-install-dialog.png)

3. Click the “Install” button next to the input bar.

```{note}

> When wanting to install a plugin from a URL you need to respect a specific syntax required by pypi!
> You need to add the vcs (version control system) information before the URL for pip to be able to handle the URL.
> You will find more information about this on the [pip documentation](https://pip.pypa.io/en/stable/topics/vcs-support/)

Reviewer: Can I suggest rephrasing this?
```suggestion
If you want to install a plugin from a URL, you need to use the required syntax (see the the [pip documentation](https://pip.pypa.io/en/stable/topics/vcs-support/) for more information.)
```

Author: Your rephrasing is really good, except that now it doesn't provide the key information that the "vcs" info needs to be added to the URL. Because the pip documentation is generic and not about how it works in the napari plugin installation process.
What do you think @melissawm ?

Reviewer 2: The provided link explains the use of `git+` and only uses examples that start like that, so I don't think we need to re-explain it here.

Author: No the link also provides similar infos for other VCS (like Git, Mercurial, Subversion  and Bazaar)

Reviewer 2: Even better :)

Author: But isn't it better to explain just what needs to be added to the url? Because the pip documentation explains the procedure for `pip install ...` but the pip install is hidden for users here, so that's why I added a short explanation here

Reviewer 2: I'm just a bit wary of adding details about *every* little thing in our guide, risking to dilute too much the information. But I don't feel strongly about this, so I'll defer to others!

Author: > 'm just a bit wary of adding details about _every_ little thing in our guide, risking to dilute too much the information.

I agree with you! I will also let more experienced napari developers decide :)

Reviewer 3: I tend to agree with @brisvag here.  I think what I would probably do in this case is to modify the second bullet point as follows:

```markdown
2. In the resulting window that opens, where it says “Install by name/URL”, enter the
    name of the plugin you want to install (or any valid pip
    [requirement specifier](https://pip.pypa.io/en/stable/reference/requirement-specifiers/) or
    [VCS scheme](https://pip.pypa.io/en/stable/topics/vcs-support))
```

which renders as:

In the resulting window that opens, where it says “Install by name/URL”, enter thename of the plugin you want to install (or any valid pip [requirement specifier](https://pip.pypa.io/en/stable/reference/requirement-specifiers/) or [VCS scheme](https://pip.pypa.io/en/stable/topics/vcs-support))

and leave it to the pip documentation to explain what valid arguments to `pip install` are

Author: I agree that is a good idea. I find also useful to have an example, a user might not know  which part of the vcs scheme is needed in the napari text bar

Reviewer 3: As in, you don't think the pip docs make that clear?

The napari bar isn't special, it literally just takes anything you can pass to `pip install`

Author: > As in, you don't think the pip docs make that clear?

Well what might not be clear is that you drop the "pip install" part and just keep the git+https... part
But maybe I'm the only one and we can keep your suggestion as is :smile:

Reviewer 3: hey @cnstt, yeah, I think i'd prefer to keep it minimal... if you want to, we could add to my example above something like "or any valid string that you would pass to `pip install` such as a requirement specifier or VCS scheme ... with the links.  If you wanna make that change we could merge this and get it in

> When wanting to install a plugin from a URL you need to respect a specific syntax required by pypi!
> You need to add the vcs (version control system) information before the URL for pip to be able to handle the URL.
> You will find more information about this on the [pip documentation](https://pip.pypa.io/en/stable/topics/vcs-support/)

Reviewer 4: ```suggestion
Napari supports installing plugins from VCS (ex. `git`, `svn`) by passing the URL to the repository in the search bar. As Napari follows the `pip` way more information about URL format may be found in [`pip` documentation](https://pip.pypa.io/en/stable/topics/vcs-support/)
```

Maybe something like this. There is a try to reduce technical details and highlight that this is only about installation from VCS, not any URL. 

This is my only suggestion, as my writing skills are not excellent.
